Assets/Mapzen/TileAddress.cs

```diff
@@ -14,16 +14,15 @@ public TileAddress(int x, int y, int z)
 
         public static TileAddress FromLngLat(LngLat lngLat, int zoom)
         {
-            double tileResolution = Geo.EarthCircumferenceMeters / Geo.TilePixelSize;
-            double tileSize = tileResolution / (1 << zoom);
+            double tileSize = Geo.EarthCircumferenceMeters / (1 << zoom);
 
             MercatorMeters meters = Geo.Project(lngLat);
 
             double px = meters.x / tileSize;
             double py = meters.y / tileSize;
 
-            int tileX = (int)(Math.Ceiling(px / Geo.TilePixelSize) - 1);
-            int tileY = (int)(Math.Ceiling(py / Geo.TilePixelSize) - 1);
+            int tileX = (int)(Math.Ceiling(px) - 1);
```

Comment:
Yes both works, `int tileX = (int)px` would perform a `floor` so it would be the same operation. In which cases could we have negative tile address? Unless we want to support world wrapping.